### PR TITLE
Add prune command

### PR DESCRIPTION
**Pull request by jasalisbury**
https://github.com/department-of-veterans-affairs/vets-contrib/issues/3133

Adds the `prune` command that removes all items of a kongfig file that have `ensure: removed`. Also removing a `yarn.lock` file that is unused and causing confusion. See https://github.com/department-of-veterans-affairs/devops/pull/5605 for an example of `prune` output.

### CODE REVIEW

File: src/cli-prune.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    remove(route.plugins, is_item_removed);
+  });
+});
+
+const yaml_config = pretty('yaml')(config);
+writeFileSync(output, yaml_config);
+
+function is_item_removed(item) {
+  return item.ensure == 'removed';
+}
```
Comment (kalilsn) [inline src/cli-prune.js]: We already have the same function in `core.js` where it's called `shouldBeRemoved`. Can you just export that one and use it here so there's no risk of them being out of sync? I could also see moving it to `utils.js` if that seems reasonable.

File: src/cli.js [hunk 2]
```diff
@@ -7,4 +7,5 @@ commander
     .allowUnknownOption()
     .command('apply', 'Apply config to a kong server', {isDefault: true})
     .command('dump', 'Dump the configuration from a kong server')
+    .command('prune', 'Remove any items marked "ensure: removed"')
```
Comment (kalilsn) [inline src/cli.js]: I could see this being read as 'Remove any items marked "ensure: removed" _from the server_'. Maybe specify `from your config file`?

**Review by kalilsn — CHANGES_REQUESTED** [review]
There are three other entity types that could be in our config files: consumers, certificates, and upstreams. We don't currently use certificates or upstreams, and we haven't yet removed a consumer, but I think we should either make this command do that or explicitly document that as a limitation.

Other than that, this is looking good! I like having this functionality in kongfig a lot!

File: src/cli.js [hunk 3]
```diff
@@ -7,4 +7,5 @@ commander
     .allowUnknownOption()
     .command('apply', 'Apply config to a kong server', {isDefault: true})
     .command('dump', 'Dump the configuration from a kong server')
+    .command('prune', 'Remove any serivce, route, global plugin or service/route plugin marked "ensure: removed" from your config file')
```
Comment (kalilsn) [inline src/cli.js]:
```suggestion
    .command('prune', 'Remove any service, route, global plugin or service/route plugin marked "ensure: removed" from your config file')
```

**Review by kalilsn — APPROVED** [review]
LGTM!

File: src/cli.js (hunk 2 above)
Comment (dvogel) [reply src/cli.js]: How about a more positive, active phrasing like _"Rewrites config, omitting any items marked "ensure: removed""_

**Review by dvogel — APPROVED** [review]

**Pull request merged**